src/context.ts

```diff
@@ -20,11 +24,29 @@ export async function resolveContext(
     const value = context[key];
 
     if (typeof value === 'function') {
-      output[key] = await value();
+      output[key] = await value(pageId);
     } else {
-      output[key] = context[key];
+      output[key] = value;
     }
   }
 
   return output;
 }
+
+export function determineContextPageId(ctx: IndexHtmlTransformContext): string {
+  // make sure paths are consistent on win32/posix
+  const fullFilePath = ctx.path.split(path.sep).join(path.posix.sep);
```

Comment:
I've seen this `normalizePath` helper in the Vite docs -- would it make sense to use that here?

https://vitejs.dev/guide/api-plugin.html#path-normalization

Reply:
I've tried to implement this, but microbundle chokes on some part in vite. 

```
microbundle -f es,cjs --target node --no-compress

Error: Unexpected token

at /Users/naranjamecanica/Developer/opensource/vite-plugin-handlebars/node_modules/vite/dist/node/chunks/dep-0776dd57.js:68846:60

68844: 
68845: const pendingModules = new Map();
68846: async function ssrLoadModule(url, server, context = { global }, urlStack = []) {                                            
```